Makefile

```diff
@@ -108,14 +114,26 @@ test-e2e-local: $(GINKGO)
 ci-e2e-kind:
 	./hack/ci-e2e-kind.sh
 
+# speed-up skaffold deployments by building all images concurrently
+export SKAFFOLD_BUILD_CONCURRENCY = 0
+extension-up extension-dev: export SKAFFOLD_DEFAULT_REPO = localhost:5001
+extension-up extension-dev: export SKAFFOLD_PUSH = true
 # use static label for skaffold to prevent rolling all gardener components on every `skaffold` invocation
-extension-up extension-down: export SKAFFOLD_LABEL = skaffold.dev/run-id=extension-local
+extension-up extension-dev extension-down: export SKAFFOLD_LABEL = skaffold.dev/run-id=extension-local
 
-extension-up: $(SKAFFOLD) $(KIND) $(HELM)
+extension-up: $(SKAFFOLD) $(KIND) $(HELM) $(KUBECTL) $(KUSTOMIZE)
```

Comment:
I now tested in local setup and `$(KUSTOMIZE)` is not needed. It is most likely using the kustomize plugin from kubectl. kubectl is enough.

See also gardener's similar targets: https://github.com/gardener/gardener/blob/5485487489b0df608413767d68560c0014c0be21/Makefile#L334-L341 . They don't require kustomize.

Drop the last commit 151c512ee960f3c653b46613f7ad06518802726a from the PR.

Reply:
I've tested it (only render and deploy) in a separate container and got `"kustomize": executable file not found in $PATH` error:
```bash
root@8cb1ccfbace9:/gardener-extension-registry-cache# make extension-up
curl -Lo /gardener-extension-registry-cache/hack/tools/bin/skaffold https://storage.googleapis.com/skaffold/releases/v2.11.1/skaffold-linux-arm64
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 79.6M  100 79.6M    0     0  8236k      0  0:00:09  0:00:09 --:--:-- 10.0M
chmod +x /gardener-extension-registry-cache/hack/tools/bin/skaffold
WARN[0000] Could not get docker context: starting command docker context inspect --format {{.Endpoints.docker.Host}}: exec: "docker": executable file not found in $PATH, falling back to the default docker host  subtask=-1 task=DevLoop
WARN[0000] failed to detect active kubernetes cluster node platform. Specify the correct build platform in the `skaffold.yaml` file or using the `--platform` flag  subtask=-1 task=DevLoop
No tags generated
Starting test...
WARN[0000] unable to get kubectl client version: starting command kubectl version --client -ojson: exec: "kubectl": executable file not found in $PATH  subtask=-1 task=DevLoop
starting command kustomize build /gardener-extension-registry-cache/local-setup: exec: "kustomize": executable file not found in $PATH
make: *** [Makefile:125: extension-up] Error 1
```